nodes/core/io/21-httpin.js

```diff
@@ -134,6 +159,16 @@ module.exports = function(RED) {
                         }
                         msg.res.set('content-length', len);
                     }
+                    
+                    // Deliberately setting AFTER content-length has been calculated as these fields will be deleted by middleware!
+                    if(RED.settings.metricsOn && RED.settings.metricsOn === true) {
```

Comment:
This is not elegant but I haven't found any other way to pass all the data required to the middleware.